stock_picking_batch_planner/models/stock_picking_batch.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2025 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import api, fields, models
+
+
+class StockPickingBatch(models.Model):
+    _inherit = "stock.picking.batch"
+
+    origin_batchable = fields.Boolean(compute="_compute_origin_batchable")
+    count_origin_batches = fields.Integer(compute="_compute_count_origin_batches")
+
+    @api.depends(
+        "picking_ids",
+        "picking_ids.move_ids.move_orig_ids",
+        "picking_ids.move_ids.move_orig_ids.picking_id",
+        "picking_ids.move_ids.move_orig_ids.move_line_ids",
+    )
+    def _compute_origin_batchable(self):
+        for record in self:
+            origin_batchable = True
+            origin_pickings = self.picking_ids.mapped(
+                "move_ids.move_orig_ids"
+            ).picking_id
+            for picking in origin_pickings:
+                if not picking._is_auto_batchable():
+                    origin_batchable = False
+                    break
+            if not origin_batchable:
+                origin_move_line_ids = origin_pickings.move_line_ids
+                for move_line in origin_move_line_ids:
+                    if not move_line._is_auto_waveable():
+                        origin_batchable = False
+                        break
+            record.origin_batchable = origin_batchable
```

Comment:
Iterate one by one and when you find a waveable case, break.

```suggestion
    def _compute_origin_batchable(self):
        for record in self:
            origin_pickings = self.picking_ids.mapped(
                "move_ids.move_orig_ids"
            ).picking_id
            # If no origin pickings, nothing can be planned
            if not origin_pickings:  
                record.origin_batchable = False
                continue
            for picking in origin_pickings:
                origin_batchable_waveable = False
                if picking._is_auto_batchable():
                    origin_batchable_waveable = True
                else:
                    for move_line in picking.move_line_ids:
                        if move_line._is_auto_waveable():
                            origin_batchable_waveable = True
                            break
                record.auto_batchable = origin_batchable_waveable
```